import_ldraw.py

```diff
@@ -318,18 +326,19 @@ def parse(self, filename):
                                 x, y, z, a, b, c,
                                 d, e, f, g, h, i
                             ) = map(float, tmpdate[2:14])
-                            mat_new = self.mat * mathutils.Matrix((
-                                    (a, b, c, x),
-                                    (d, e, f, y),
-                                    (g, h, i, z),
-                                    (0, 0, 0, 1)
-                                ))
-
+                            # - Reset orientation of top-level Part, track original orientation
+                            if self.part_count == 1 and isPart:
+                                mat_new = self.mat * mathutils.Matrix(((1, 0, 0, 0), (0, 1, 0, 0), (0, 0, 1, 0), (0, 0, 0, 1)))
+                                mat_ref = self.mat * mathutils.Matrix(((a, b, c, x), (d, e, f, y), (g, h, i, z), (0, 0, 0, 1))) * mathutils.Matrix.Rotation(math.radians(90), 4, 'X')
+                            else:
+                                mat_new = self.mat * mathutils.Matrix(((a, b, c, x), (d, e, f, y), (g, h, i, z), (0, 0, 0, 1)))
                             color = tmpdate[1]
                             if color == '16':
                                 color = self.colour
                             subfiles.append([new_file, mat_new, color])
-
+                            # - When top-level Part, save orientation separately
+                            if self.part_count == 1 and isPart:
```

Comment:
Aha, I feel a challenge coming up. As I mentioned before, this top-level brick definition is really the essence for brick linking to work correct. I'm going to study this further.